ipatests/test_integration/test_commands.py

```diff
@@ -699,3 +699,61 @@ def is_tls_version_enabled(tls_version):
         assert not is_tls_version_enabled('tls1')
         assert not is_tls_version_enabled('tls1_1')
         assert is_tls_version_enabled('tls1_2')
+
+    def test_sss_ssh_authorizedkeys(self):
+        """
+        When there is ssh key is added for IPA user then
+        'sss_ssh_authorizedkeys ipauser' should return ssh key.
+
+        Test for : https://pagure.io/SSSD/sssd/issue/3937
+
+        Steps:
+        1. add ipa user with ssh-key.
+        2. call sss_ssh_authorizedkeys and ssh key should be returned
+        3. check user is able to ssh or not.
+        """
+        user = 'testsshuser'
+        passwd = 'Secret123'
+        user_key = '/tmp/ssh-key'
```

Comment:
It is better to create temp file using mktemp to avoid collisions with existing files.
There is a helper `tasks.create_temp_file` to create temp file on remote host.

Same applies to pem_file variable

Reply:
For ssh-key using it as ssh-key{0}".format(random.randint(1, 99)) 
If i use tasks.create_temp_file it will throw error for ssh-keygen that file already exist / or you want to override. 
so tasks.create_temp_file is using only for  pem_file